managed/src/SwiftlyS2.Core/Services/PluginConfigurationService.cs

```diff
@@ -96,21 +226,55 @@ public IPluginConfigurationService InitializeWithTemplate(string name, string te
     return this;
   }
 
-  public IPluginConfigurationService Configure(Action<IConfigurationBuilder> configure) {
+  public IPluginConfigurationService InitializeTomlWithModel<T>(string name, string sectionName) where T : class, new()
+  {
+
+    var configPath = GetConfigPath(name);
+
+    if (File.Exists(configPath))
+    {
+      return this;
+    }
+
+    var dir = Path.GetDirectoryName(configPath);
+    if (dir is not null)
+    {
+      Directory.CreateDirectory(dir);
+    }
+
+    var config = new T();
+
+    // Create a TomlTable and add the section
+    var tomlTable = new TomlTable();
+    var sectionTable = TomlModelHelper.ObjectToTomlTable(config);
```

Comment:
doesnt Toml.FromModel do this already? It takes in an object and spits out a string. Or is there some specific reason why TomlTable is needed here?